### PR TITLE
[DO NOT MERGE] feat: use import to run wasm/client code

### PR DESCRIPTION
**Pull request by JayWhite2357**
# Rationale for this change

We want to be able to use the SDK more simply from within Chainlink.

# What changes are included in this PR?

See the individual commits.

In particular, there is a `build_index.sh` script that prepends the generated js code from wasm-pack to the index_tail.js. None of the generated code needs to change. It can simply be used from within the index_tail.js code.

Ultimately, we need to turn this script into a CI job that publishes to npm.

# Are these changes tested?
Yes.

### CODE REVIEW

File: chainlink-functions/source/build_index.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+cd $SCRIPT_DIR
+
+curl $1/sxt_proof_of_sql_sdk_wasm.js | cat - ./index_tail.js > ./index.js
+curl $1/sxt_proof_of_sql_sdk_wasm_bg.wasm > ./sxt_proof_of_sql_sdk_wasm_bg.wasm
```
Comment (Dustin-Ray) [inline chainlink-functions/source/build_index.sh:6]: NIT: newline

File: chainlink-functions/source/index_tail.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  const id = setTimeout(() => controller.abort(), 3000);
+  const response = await fetch(url, {
+    method: "POST",
+    headers,
+    body: data ? JSON.stringify(data) : undefined,
+    signal: controller.signal,
+  });
+  clearTimeout(id);
+  return response;
+}
```
Comment (Dustin-Ray) [inline chainlink-functions/source/index_tail.js:103]: How do we know that this will succeed for all request types? There is a lot left out here from the Chainlink version of this function. What if this example needs to retrieve binary data at some point?

**Comment by Dustin-Ray** [comment]
This branch currently fails with:

```
Start simulation...
Simulation result {
  capturedTerminalOutput: '\x1B[0m\x1B[32mDownload\x1B[0m https://raw.githubusercontent.com/spaceandtimelabs/sxt-proof-of-sql-sdk/feat/import-sdk/chainlink-functions/source/index.js\n' +
    'Workflow failed:  verification failure: failed to interpret or verify query results: Verification error: round evaluation does not match claimed sum\n',
  responseBytesHexstring: '0x4661696c65643a20'
}
✅ Decoded response to uint256:  5071450569674340896n
```

Is this expected?

**Comment by Dustin-Ray** [comment]
I also noticed the cost of running this example has ballooned in the simulation, although its unlikely this is the real world cost, just worth pointing out:

```
Estimate request costs...
Fulfillment cost estimated to 76.237712263510214221 LINK
```

File: chainlink-functions/source/sxt_proof_of_sql_sdk_wasm_bg.wasm
Comment (tlovell-sxt) [inline chainlink-functions/source/sxt_proof_of_sql_sdk_wasm_bg.wasm:1]: Should this be committed? The wasm-packed js isn't

File: chainlink-functions/example/source.js [hunk 3]
```diff
@@ -1,589 +1,33 @@
...
+  "https://raw.githubusercontent.com/spaceandtimelabs/sxt-proof-of-sql-sdk/feat/import-sdk/chainlink-functions/source/index.js"
+);
 
-// TODO: make this not be hardcoded:
+// Define test parameters
+const queryString = "SELECT SUM(BLOCK_NUMBER), COUNT(*) FROM ETHEREUM.BLOCKS";
+const table = "ETHEREUM.BLOCKS";
+// TODO: This is currently hardcoded. But, we need to make it dynamic.
 const commitmentKey =
   "0xca407206ec1ab726b2636c4b145ac28749505e273536fae35330b966dac69e86a4832a125c0464e066dd20add960efb518424c4f434b5320455448455245554d4a9e6f9b8d43f6ad008f8c291929dee201";
```
Comment (Dustin-Ray) [inline chainlink-functions/example/source.js:12]: We dont need to hard code this anymore but maybe that can be a different PR

**Comment by JayWhite2357** [comment]
This PR is not ready to merge. The majority of it should never be merged into main.
